dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorRecordStore.cs

```diff
@@ -359,15 +359,12 @@ private static async Task<T> RunOperationAsync<T>(string collectionName, string
         }
         catch (RedisConnectionException ex)
         {
-            var wrapperException = new VectorStoreOperationException("Call to vector store failed.", ex);
-
-            // Using Open Telemetry standard for naming of these entries.
-            // https://opentelemetry.io/docs/specs/semconv/attributes-registry/db/
-            wrapperException.Data.Add("db.system", DatabaseName);
-            wrapperException.Data.Add("db.collection.name", collectionName);
-            wrapperException.Data.Add("db.operation.name", operationName);
-
-            throw wrapperException;
+            throw new VectorStoreOperationException("Call to vector store failed.", ex)
+            {
+                DBSystem = DatabaseName,
+                DBCollectionName = collectionName,
+                DBOperationName = operationName
```

Comment:
1. Are we going to add logging on our side? If yes, fields like `db.system` and `db.colleciton.name` will still be applicable, but only for telemetry side of things.
2. Taking into account that we are using strongly typed exception and fields, do we need to preserve OpenTelemetry naming in exception properties? Because we use `VectorStore` naming around the code, but here we say `DBSystem`, which may be confusing. I would name them something like `VectorStoreName`, `CollectionName`, `OperationName`.

Reply:
Thanks for the feedback @dmytrostruk, that makes sense to me.  I'll update the names.